torch/autograd/function.py

```diff
@@ -533,12 +534,29 @@ def vmap(info, in_dims, *args):
 
     @classmethod
     def apply(cls, *args, **kwargs):
+        def bind_default_args(func, is_def_ctx, *args, **kwargs):
+            arguments: List[Any]
+
+            arguments = list(args)
+            # Used to bind to ctx
+            if is_def_ctx:
+                arguments.insert(0, None)
+
+            signature = inspect.signature(func)
+            bound_args = signature.bind(*arguments, **kwargs)
+            bound_args.apply_defaults()
+
+            return bound_args.args[1:] if is_def_ctx else bound_args.args
+
+        is_def_ctx = cls.setup_context == _SingleLevelFunction.setup_context
```

Comment:
I keep getting confused about whether the def here means "default" or "defined". 
Could we do something like `is_setup_ctx_defined = cls.setup_context != _SingleLevelFunction.setup_context`

Reply:
Updated it.